src/openrct2/peep/peep.c

```diff
@@ -9823,10 +9823,8 @@ static sint32 guest_path_find_leaving_park(rct_peep *peep, rct_map_element *map_
 	rct2_peep_spawn* peepSpawn = &gPeepSpawns[0];
 	// Peeps for whatever reason return to their original spawn point
 	// this in future should look for the nearest.
-	if (peep->sprite_index & 1 && gPeepSpawns[1].x != PEEP_SPAWN_UNDEFINED) {
-		peepSpawn++;
-	}
-
+	get_random_peep_spawn(peepSpawn);
```

Comment:
I actually just whipped together a function that grabs the nearest peep spawn based on distance and square rooting, I'm about to test it real quick and see how well it works. if it works decent, I'll provide a copy of the forked park entrance peep spawn i have for megaworld park, which has 16 park entrance paths.